go/common/tribool/tribool_test.go

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) 2024 Fantom Foundation
+//
+// Use of this software is governed by the Business Source License included
+// in the LICENSE file and at fantom.foundation/bsl11.
+//
+// Change Date: 2028-4-16
+//
+// On the date above, in accordance with the Business Source License, use of
+// this software will be governed by the GNU Lesser General Public License v3.
+
+package tribool
+
+import "testing"
+
+func TestTribool(t *testing.T) {
+	tests := []struct {
+		name string
+		tb   Tribool
+		want struct {
+			unknown  bool
+			trueVal  bool
+			falseVal bool
+		}
+	}{
+		{
+			name: "UnknownTribool",
+			tb:   NewUnknown(),
+			want: struct {
+				unknown  bool
+				trueVal  bool
+				falseVal bool
+			}{true, false, false},
+		},
+		{
+			name: "FalseTribool",
+			tb:   NewFalse(),
+			want: struct {
+				unknown  bool
+				trueVal  bool
+				falseVal bool
+			}{false, false, true},
+		},
+		{
+			name: "TrueTribool",
+			tb:   NewTrue(),
+			want: struct {
+				unknown  bool
+				trueVal  bool
+				falseVal bool
+			}{false, true, false},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got, want := tt.tb.Unknown(), tt.want.unknown; got != want {
+				t.Errorf("got: Unknown() = %v, want %v", got, want)
+			}
+			if got, want := tt.tb.True(), tt.want.trueVal; got != want {
+				t.Errorf("got: True() = %v, want %v", got, want)
+			}
+			if got, want := tt.tb.False(), tt.want.falseVal; got != want {
+				t.Errorf("got: False() = %v, want %v", got, want)
+			}
+		})
+	}
+}
```

Comment:
I think it would also be important to have a test that the `==` and `!=` operator work as expected. This should also allow them to be keys in maps, be automatically deep-copied in slices, ...

Reply:
right, will add this test